### PR TITLE
ECVRF key pair type

### PR DESCRIPTION
**Pull request by gdbelvin**

### CODE REVIEW

File: go/vrf/ecvrf.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	d *big.Int
+}
+
+// Public returns the public key corresponding to priv.
+func (priv *PrivateKey) Public() *PublicKey {
+	return &priv.PublicKey
+}
+
+func NewKey(curve elliptic.Curve, sk []byte) *PrivateKey {
+	Yx, Yy := curve.ScalarBaseMult(sk)
```
Comment (cindylindeed) [inline go/vrf/ecvrf.go]: Why Yx and Yy and not just X, Y? It is not clear from https://golang.org/src/crypto/elliptic/elliptic.go?s=7072:7143#L269.

File: go/vrf/ecvrf.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	return &PrivateKey{
+		d:         new(big.Int).SetBytes(sk),             // Use SK to derive the VRF secret scalar x
+		PublicKey: PublicKey{Curve: curve, X: Yx, Y: Yy}, // VRF public key Y = x*B
```
Comment (cindylindeed) [inline go/vrf/ecvrf.go]: nit: Reverse order of d and PublicKey to match the struct definition on :29.

File: go/vrf/ecvrf.go (hunk 1 above)
Comment (gdbelvin) [reply go/vrf/ecvrf.go]: done

File: go/vrf/ecvrf.go (hunk 2 above)
Comment (gdbelvin) [reply go/vrf/ecvrf.go]: done

**Review by cindylindeed — APPROVED** [review]

**Pull request merged**